src/tyr/matrix_serializer.cc

```diff
@@ -118,6 +145,17 @@ json::ArrayPtr serialize_row(const valhalla::Matrix& matrix,
       if (!date_time.empty()) {
         map->emplace("date_time", date_time);
       }
+      // TODO(nils): shapes aren't implemented yet in TDMatrix
+      if (matrix.shapes().size() && !matrix.shapes()[i].empty()) {
```

Comment:
i question this, why isnt:

```suggestion
      if (matrix.shapes().size()) {
```

and then further down we emplace a `nullptr` when its empty

Reply:
my assumption is that size == 0 when you didnt want shape and that if you have an empty entry its because of a disconnected pair (which we still need column form in the row)